### PR TITLE
Ignore .gitignores in ignored paths (#1785)

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/working_copy.rs [hunk 1]
```diff
@@ -680,19 +681,59 @@ impl TreeState {
...
-                            && current_tree.path_value(&sub_path).is_none()
-                        {
-                            continue;
+                        if git_ignore.matches_all_files_in(&sub_path.to_internal_dir_string()) {
+                            // If the whole directory is ignored, visit only paths we're already
+                            // tracking.
+                            let tracked_paths = self
+                                .file_states
+                                .range((Bound::Excluded(&sub_path), Bound::Unbounded))
+                                .take_while(|(sub_path, _)| sub_path.contains(sub_path))
```
Comment (yuja) [inline lib/src/working_copy.rs:690]: `sub_path` shadows the outer `sub_path`

**Review by yuja — APPROVED** [review]

File: lib/src/working_copy.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/working_copy.rs:690]: I had renamed the outer one to `path` for a while then renamed it back to `sub_path`. I'm glad you noticed. I'll rename the outer one back to `path`.

Comment (martinvonz) [reply lib/src/working_copy.rs]: That's now done. I inserted a commit before the last one, in case you want to take another look before I merge this.

**Review by yuja — APPROVED** [review]
LGTM, thanks.

**Pull request merged**